src/Arcus.Security.Providers.AzureKeyVault/Arcus.Security.Providers.AzureKeyVault.csproj

```diff
@@ -23,14 +23,16 @@
     <None Include="..\..\docs\static\img\icon.png" Pack="true" PackagePath="\"/>
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0'">
-    <PackageReference Include="Guard.Net" Version="2.0.0" />
+  <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.Configuration" Version="6.0.0" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' != 'net6.0'">
+  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0' or '$(TargetFramework)' == 'net8.0'">
+    <PackageReference Include="Guard.Net" Version="2.0.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'net6.0' and '$(TargetFramework)' != 'net8.0'">
     <PackageReference Include="Guard.Net" Version="1.2.0" />
```

Comment:
Same as another comment.   I think this is strange.  
.NET 6 and .NET 8 should implement .NET Standard 2.1, so I think it is strange that we need to use another version of Guard here for .NET standard ?

Reply:
![image](https://github.com/arcus-azure/arcus.security/assets/9039753/d7111bdd-166f-4bca-b81f-5b86faf538ad)
v2 is not compatible, but v3 is. Maybe we should then move to v3, yes?

Reply:
Also want to remove those obsolete packages in this new major version, so we fully rely on the `Azure.*` packages.

Reply:
Updated to Guard v3 here: https://github.com/arcus-azure/arcus.security/pull/421/commits/d24b66901b7efe25e896f007b5b6a323722a58c4